mobile_verifier/src/coverage.rs

```diff
@@ -0,0 +1,719 @@
+                INSERT INTO hex_coverage
+                  (uuid, hex, indoor, cbsd_id, signal_level, coverage_claim_time, inserted_at)
+                VALUES
+                  ($1, $2, $3, $4, $5, $6, $7)
+                "#,
+            )
+            .bind(self.uuid)
+            .bind(location as i64)
+            .bind(self.indoor)
+            .bind(&self.cbsd_id)
+            .bind(SignalLevel::from(hex.signal_level))
+            .bind(self.coverage_claim_time)
+            .bind(Utc::now())
+            .execute(&mut *transaction)
+            .await?;
+        }
+
+        Ok(true)
+    }
+}
+
+async fn validate_coverage_object(
+    coverage_object: &CoverageObjectIngestReport,
+    auth_client: &mut AuthorizationClient,
+) -> anyhow::Result<CoverageObjectValidity> {
+    if !auth_client
+        .verify_authorized_key(&coverage_object.report.pub_key, NetworkKeyRole::MobilePcs)
+        .await?
+    {
+        return Ok(CoverageObjectValidity::InvalidPubKey);
+    }
+
+    Ok(CoverageObjectValidity::Valid)
+}
+
+#[derive(Clone, FromRow)]
+pub struct HexCoverage {
+    pub uuid: Uuid,
+    pub hex: i64,
+    pub indoor: bool,
+    pub cbsd_id: String,
+    pub signal_level: SignalLevel,
+    pub coverage_claim_time: DateTime<Utc>,
+}
+
+#[derive(Eq)]
+struct CoverageLevel {
+    cbsd_id: String,
+    coverage_claim_time: DateTime<Utc>,
+    hotspot: PublicKeyBinary,
+    indoor: bool,
+    signal_level: SignalLevel,
+}
+
+impl PartialEq for CoverageLevel {
+    fn eq(&self, other: &Self) -> bool {
+        self.coverage_claim_time == other.coverage_claim_time
+    }
+}
+
+impl PartialOrd for CoverageLevel {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.coverage_claim_time.cmp(&other.coverage_claim_time))
+    }
+}
+
+impl Ord for CoverageLevel {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.coverage_claim_time.cmp(&other.coverage_claim_time)
+    }
+}
+
+impl CoverageLevel {
+    fn coverage_points(&self) -> anyhow::Result<Decimal> {
+        Ok(match (self.indoor, self.signal_level) {
+            (true, SignalLevel::High) => dec!(400),
+            (true, SignalLevel::Low) => dec!(100),
+            (false, SignalLevel::High) => dec!(16),
+            (false, SignalLevel::Medium) => dec!(8),
+            (false, SignalLevel::Low) => dec!(4),
+            (_, SignalLevel::None) => dec!(0),
+            _ => anyhow::bail!("Indoor radio cannot have a signal level of medium"),
+        })
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub struct CoverageReward {
+    pub cbsd_id: String,
+    pub points: Decimal,
+    pub hotspot: PublicKeyBinary,
+}
+
+pub const MAX_RADIOS_PER_HEX: usize = 5;
+
+#[async_trait::async_trait]
+pub trait CoveredHexStream {
+    async fn covered_hex_stream<'a>(
+        &'a self,
+        cbsd_id: &'a str,
+        coverage_obj: &'a Uuid,
+        period_end: DateTime<Utc>,
+    ) -> Result<BoxStream<'a, Result<HexCoverage, sqlx::Error>>, sqlx::Error>;
```

Comment:
are these errors overly prescriptive for a trait? why have the trait at all if it always assumes a postgres/sql source?